plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogConnectorSmokeTest.java

```diff
@@ -184,6 +194,245 @@ public void testCommentViewColumn()
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')");
+
+        String showCreateTableOld = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+        String tableLocation = getTableLocation(tableName);
+
+        // Drop table from glue metastore and use the same table name to register again with the metadata
+        dropTableFromMetastore(tableName);
+
+        String tableNameNew = "test_register_table_with_different_table_name_new_" + icebergFileFormat + "_" + randomTableSuffix();
+
+        assertUpdate(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", schemaName, tableNameNew, tableLocation));
+        String showCreateTableNew = (String) computeActual("SHOW CREATE TABLE " + tableNameNew).getOnlyValue();
+
+        assertUpdate(format("INSERT INTO %s values(3, 'POLAND', true)", tableNameNew), 1);
+        assertThat(query(format("SELECT * FROM %s", tableNameNew)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')," +
+                        " ROW(INT '3', VARCHAR 'POLAND', BOOLEAN 'true')");
+        assertThat(showCreateTableOld.replaceFirst(tableName, tableNameNew)).isEqualTo(showCreateTableNew);
+        assertUpdate(format("DROP TABLE %s", tableNameNew));
+    }
+
+    @Test
+    public void testRegisterTableWithDroppedTable()
+    {
+        IcebergFileFormat icebergFileFormat = ORC;
+        String tableName = "test_register_table_with_dropped_table_" + icebergFileFormat + "_" + randomTableSuffix();
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean) with (format = '%s')", tableName, icebergFileFormat));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", tableName), 1);
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')");
+
+        String tableLocation = getTableLocation(tableName);
+
+        // Drop table to verify register_table call fails when no metadata can be found (table doesn't exist)
+        assertUpdate(format("DROP TABLE %s", tableName));
+
+        String tableNameNew = "test_register_table_with_invalid_table_location_and_metadata_file_new_" + icebergFileFormat + "_" + randomTableSuffix();
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", schemaName, tableNameNew, tableLocation),
+                ".*No versioned metadata file exists at location.*");
+    }
+
+    @Test
+    public void testRegisterTableWithNoMetadataFile()
+    {
+        // TODO find the way to use filesystem api then implement the test
```

Comment:
Here is a bit of inspiration

https://github.com/trinodb/trino/blob/6e29d60fd99370e197739bf74e00cb20725bb4d4/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogConnectorSmokeTest.java#L88-L92

It is a bit more low-level than `TrinoFileSystem`, but it should fit your needs.

Reply:
Actually, initializing `fileSystem` field in `TestIcebergGlueCatalogConnectorSmokeTest` constructor will also do just fine:

```
fileSystem = new HdfsFileSystemFactory(HDFS_ENVIRONMENT).create(TestingConnectorSession.SESSION);
```

Reply:
Thanks! I am able to use `HDFS_ENVIRONMENT`. Remaining tests are implemented.